### PR TITLE
fix: carousel editor & ui fixes

### PR DESCRIPTION
**Pull request by cwaring**
Small ui/logic fixes discussed on Slack. (cherry-picked from #575)

### CODE REVIEW

File: components/core/CarouselSidebarData.js [hunk 1]
```diff
@@ -472,20 +474,20 @@ class CarouselSidebarData extends React.Component {
       <div key="s-2" style={{ marginBottom: 80 }}>
         <div css={STYLES_META}>
           {this.state.isEditing ? (
-            <Boundary enabled onOutsideRectEvent={this._handleEditFilename}>
+            <Boundary enabled onOutsideRectEvent={this._handleEditFilename.bind(this)}>
```
Comment (jimmylee) [inline components/core/CarouselSidebarData.js:477]: You don't need to `.bind(this)` if you're using an arrow function.

File: components/core/CarouselSidebarData.js [hunk 2]
```diff
@@ -472,20 +474,20 @@ class CarouselSidebarData extends React.Component {
...
               <Input
                 full
                 value={this.state.name}
                 name="name"
-                onChange={this._handleChange}
+                onChange={this._handleChange.bind(this)}
```
Comment (jimmylee) [inline components/core/CarouselSidebarData.js:482]: Same here `You don't need to `.bind(this)` if you're using an arrow function.`

File: components/core/CarouselSidebarData.js [hunk 3]
```diff
@@ -412,7 +413,8 @@ class CarouselSidebarData extends React.Component {
     }
   };
 
-  _handleEditFilename = async () => {
+  _handleEditFilename = () => {
```
Comment (jimmylee) [inline components/core/CarouselSidebarData.js:416]: No bind is necessary.

**Review by jimmylee — CHANGES_REQUESTED** [review]

**Comment by cwaring** [comment]
Super strange, there was a case yesterday where `this` was pointing to a closure function and unable to access the class properties which was causing these bugs, I suspected it was because these elements were being pushed into an array before rendering. Now I've restarted my env and I'm unable to recreate it 🤷‍♂️

I'll close this for now and file it under babel weirdness.

**Pull request closed without merging**